spec/controllers/post_spec.rb

```diff
@@ -37,7 +37,24 @@
       expect(assigns(:posts)).to match_array([post1])
     end
 
-    it "searches for description" do
+    it "searches by title" do
+      post1 = create(
+        :post,
+        title: "Xyz",
+        show: true
+      )
+
+      create(
+        :post,
+        show: true
+      )
+
+      get :index, query: "xyz"
```

Comment:
I prefer `title = 'Xyz'` -> create post with title, `get :index query title` to ensure they're the same. Seeing case difference here. Also maybe make the other post have a different title rather than a default value so it makes the filter more explicit in the test.